features/step_definitions/backend_api_steps.rb

```diff
@@ -39,6 +39,10 @@
   @product.backend_api_configs.create!(backend_api: @backend, path: "/my_product")
 end
 
+Given "the backend is used by this service" do
+  @service.backend_api_configs.create!(backend_api: @backend, path: "/my_product")
+end
+
```

Comment:
You've duplicated the same step from above. Use that instead.

Reply:
Not really, because the first modifies `@product` and the second modifies `@service`

Reply:
Same logic and model exactly, different nomenclature. Moreover, service is outdated and we should use product. Define @product in https://github.com/3scale/porta/pull/3271/files#diff-c9c27f4519be20d65e92d0aad9b8195f25e1f62748ff556399e9c6e04da9e965R8 or do double assignment `@service = @product = ...`

Reply:
And if you reeeally want to change the name then you could do a polyvalent step such as:
```
Given "the backend is used by this service/product" do
  (@service || @product) ...
end
```

Reply:
Also, in no case would make sense to have both `@service` and `@product` defined in the same Scenario, because they refer to the same instance.

Reply:
all over the application both service and product are used 
even if you go to product list also the route says service so I think we can use both.

Reply:
May be we can use double assignment

Reply:
TL;DR: don't use service. Use product.

This all come from the times before API as a Product (APIAP).

Back in the day, API and service was the same, that's why you see _service_ all over the tests. However, since the implementation of APIAP we have Product and Backend.
Service and Product are the same: `Service` is still the model but the "concept" was renamed to Product, because technically BackendAPIs are the services. Therefore, we should quit using `service` and use `product` and `backend` instead, specially when we use it in a "public context" such as a cucumber scenario that is part of the specification / documentation.